apollo-converters/pojo/src/main/java/com/apollostack/converter/pojo/ApolloResponseBodyConverter.java

```diff
@@ -0,0 +1,107 @@
+package com.apollostack.converter.pojo;
+
+import com.apollostack.api.graphql.Error;
+import com.apollostack.api.graphql.Operation;
+import com.apollostack.api.graphql.Response;
+import com.apollostack.api.graphql.ResponseStreamReader;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Type;
+import java.util.List;
+
+import okhttp3.ResponseBody;
+import retrofit2.Converter;
+
+class ApolloResponseBodyConverter implements Converter<ResponseBody, Response<? extends Operation.Data>> {
+  private final Type type;
+
+  ApolloResponseBodyConverter(Type type) {
+    this.type = type;
+  }
+
+  @Override public Response<? extends Operation.Data> convert(ResponseBody value) throws IOException {
+    BufferedSourceJsonReader jsonReader = new BufferedSourceJsonReader(value.source());
+    jsonReader.beginObject();
+
+    ResponseStreamReader responseStreamReader = new ResponseJsonStreamReader(jsonReader);
+    Operation.Data data = null;
+    List<Error> errors = null;
+    while (responseStreamReader.hasNext()) {
+      String name = responseStreamReader.nextName();
+      if ("data".equals(name)) {
+        data = readResponseData(responseStreamReader);
+      } else if ("errors".equals(name)) {
+        errors = readResponseErrors(responseStreamReader);
+      } else {
+        responseStreamReader.skipNext();
+      }
+    }
+    jsonReader.endObject();
+
+    return new Response<>(data, errors);
+  }
+
+  private Operation.Data readResponseData(ResponseStreamReader reader) throws IOException {
+    return reader.nextObject(new ResponseStreamReader.NestedReader<Operation.Data>() {
+      @Override public Operation.Data read(ResponseStreamReader reader) throws IOException {
+        //noinspection TryWithIdenticalCatches
+        try {
+          return (Operation.Data) ((Class<?>) type).getConstructor(ResponseStreamReader.class).newInstance(reader);
```

Comment:
I have no problem with that. Mostly the cost is high when you need to reflect over dozens of field names.  One is in the ballpark of 30ms on slowest of devices.